apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginDefine.java

```diff
@@ -0,0 +1,57 @@
+package org.skywalking.apm.agent.core.plugin;
+
+import java.util.List;
+import org.skywalking.apm.util.StringUtil;
+
+public class PluginDefine {
+    /**
+     * Plugin name.
+     */
+    private String name;
+
+    /**
+     * The class name of plugin defined.
+     */
+    private String defineClass;
+
+    private PluginDefine() {
+    }
+
+    private PluginDefine(String name, String defineClass) {
+        this.name = name;
+        this.defineClass = defineClass;
+    }
+
+    public static PluginDefine build(String define) {
+        if (StringUtil.isEmpty(define)) {
+            return IllegalPluginDefine.INSTANCE;
```

Comment:
Still don't understand this enum class, also strange name `IllegalPluginDefine`. If it is illegal, why don't throw exception?